#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_interactive_marker'

from dynamic_reconfigure.parameter_generator_catkin import *

from math import pi

gen = ParameterGenerator ()

gen.add("frame_id", str_t, 0, "frame_id", "")
gen.add("parent_frame_id", str_t, 0, "parent_frame_id", "")

gen.add("x", double_t, 0, "Translation along x-axis", 0.0, -1000000.0, 1000000.0)
gen.add("y", double_t, 0, "Translation along y-axis", 0.0, -1000000.0, 1000000.0)
gen.add("z", double_t, 0, "Translation along z-axis", 0.0, -1000000.0, 1000000.0)

gen.add("roll", double_t, 0, "Rotation around x-axis", 0, -pi, pi)
gen.add("pitch", double_t, 0, "Rotation around y-axis", 0, -pi, pi)
gen.add("yaw", double_t, 0, "Rotation around z-axis", 0, -pi, pi)

exit (gen.generate (PACKAGE, PACKAGE, "TransformParameter"))
